src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -147,14 +149,168 @@ template<class T> bool pointFrustum(const Vector3<T>& point, const Frustum<T>& f
+*/
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, T radius, const Matrix4<T>& coneView, Rad<T> angle);
+
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, T radius, const Matrix4<T>& coneView, T sinAngle, T cosAngle, T tanAngle);
+
+/**
+@brief Intersection of a sphere and a cone
+@param sphereCenter Sphere center
+@param radius Sphere radius
+@param origin Origin of the cone
+@param normal Cone normal
+@param angle Cone opening angle (0 < angle < pi).
+
+Offsets the cone plane by @f$ -r\sin{\theta} \cdot \boldsymbol n @f$ (with  @f$ \theta @f$
+the cone's half-angle) which separates two half-spaces:
+In front of the plane, in which the sphere cone intersection test is equivalent
+to testing the sphere's center against a similarly offset cone (which is equivalent
+the cone with surface expanded by @f$ r @f$ in surface normal direction), and
+begind the plane, where the test is equivalent to testing whether the origin of
+the original cone intersects the sphere.
+
+Returns @cpp true @ce if the sphere intersects with the cone.
+*/
+template<class T> bool sphereCone(const Vector3<T>& sphereCenter, const T radius, const Vector3<T>& origin, const Vector3<T>& normal, const Rad<T> angle);
+
+template<class T> bool sphereCone(const Vector3<T>& sphereCenter, const T radius, const Vector3<T>& origin, const Vector3<T>& normal, const T sinAngle, const T tanAngleSquaredPlusOne);
+
+/**
+@brief Intersection of an axis aligned bounding box and a cone
+
+@param center Center of the AABB
+@param extents (Half-)Extents of the AABB
+@param origin Origin of the cone
+@param normal Normal of the cone
+@param angle Opening angle of the cone
+
+On each axis finds the intersection points of the cones axis with infinite planes
+obtained by extending the two faces of the box that are perpendicular to that axis.
+
+The intersection points on the planes perpendicular to axis @f$ a \in {0, 1, 2} @f$
+are given by @f[
+    \boldsymbol i = \boldsymbol n \cdot \frac{(\boldsymbol c_a - \boldsymbol o_a) \pm \boldsymbol e_a}{\boldsymbol n_a}
+@f]
+
+with normal @f$ n @f$, cone origin @f$ o @f$, box center @f$ x @f$ and box extents @f$ e @f$.
+
+The points on the faces that are closest to this intersection point are the closest
+to the cones axis and are tested for intersection with the cone using
+@ref pointCone(const Vector3<T>&, const Vector3<T>&, const Vector3<T>&, const T)
+
+As soon as an intersecting point is found, the function returns @cpp true @ce.
+If all points lie outside of the cone, it will return @cpp false @ce.
+
+Returns @cpp true @ce if the axis aligned bounding box intersects the cone.
+*/
+template<class T> bool aabbCone(const Vector3<T>& center, const Vector3<T>& extents, const Vector3<T>& origin, const Vector3<T>& normal, const Rad<T> angle);
+
+template<class T> bool aabbCone(const Vector3<T>& center, const Vector3<T>& extents, const Vector3<T>& origin, const Vector3<T>& normal, const T tanAngleSquaredPlusOne);
```

Comment:
Since the documentation is a very important part of this, it would be great if you could generate it and verify it says what it should say.

For the overloads, a nice thing could be to say how all the extra parameters are precomputed and also referencing them from the convenience versions.